client/blocks/post-share/connection.jsx

```diff
@@ -47,9 +49,11 @@ const PostShareConnection = ( { connection, isActive, onToggle } ) => {
 			</div>
 
 			<div className="post-share__service-account-name">
-				<span>{ external_display }</span>
+				<label id={ id }>{ external_display }</label>
```

Comment:
Optional and material for follow-up: it would be even better if we somehow include the service name too for the label (LinkedIn, Facebook, Twitter, etc.), but without any visual changes.